services/src/main/java/org/keycloak/services/clientpolicy/executor/impl/AbstractClientPoicyExecutor.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.services.clientpolicy.executor.impl;
+
+import org.jboss.logging.Logger;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutor;
+import org.keycloak.services.clientregistration.ClientRegistrationContext;
+import org.keycloak.services.clientregistration.policy.RegistrationAuth;
+import org.keycloak.services.resources.admin.AdminAuth;
+
+/**
+ * Executor can override the client settings to enforce some actions.
+ * This feature can be activated or deactivated.
+ */
+public abstract class AbstractClientPoicyExecutor implements ClientPolicyExecutor {
```

Comment:
> For update events the augment for all executors needs to be done prior to the isSatisfied as it could be that one executor adds something that a different executor requires.

One point I'd like to confirm it is that an executor do augment and validation against clients that satisfies all conditions of a policy. Therefore, I intended that no method of an executor is called before `isSatisfied...()` of a condition are called. Is that correct?

Reply:
Yes, it has to be that way